src/lib/spinel/CMakeLists.txt

```diff
@@ -26,27 +26,22 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
+add_library(openthread-radio-spinel)
 add_library(openthread-spinel-ncp)
 add_library(openthread-spinel-rcp)
 
 target_compile_definitions(openthread-spinel-ncp PRIVATE
-    OPENTHREAD_FTD=1
```

Comment:
@zhanglongxia in commit e8b6114f8c3d1f60d797b73ea7aadd988c7711f2 which is part of this PR you removed this line as well as the lines that set `OPENTHREAD_CONFIG_NCP_HDLC_ENABLE` below and the removal of `OPENTHREAD_RADIO` when building `openthread-spinel-rcp`.  The commit message indicates that these are "useless configurations in CMake file" but in my searches I don't think they are useless.  For example, `OPENTHREAD_CONFIG_NCP_HDLC_ENABLE` needs to be set properly when compiling `ncp_hdlc.cpp`.  I could very well be missing how its being set some place else.  Can you help me understand why these definitions were removed.  Thanks.

Reply:
The `useless configurations in CMake file` means the configurations are useless in the file `src/lib/spinel/CMakeLists.txt`. Because there are no files that under the path `src/lib/spinel` use the `OPENTHREAD_CONFIG_NCP_HDLC_ENABLE `. 

The file `ncp_hdlc.cpp` is under `src/ncp`, the `OPENTHREAD_CONFIG_NCP_HDLC_ENABLE` for `ncp_hdlc.cpp` is configured by [src/ncp/CMakeLists.txt#L58](https://github.com/zhanglongxia/openthread/blob/master/src/ncp/CMakeLists.txt#L58)

Reply:
Thank you @zhanglongxia . That make sense.